### PR TITLE
Feature/tri 344

### PR DESCRIPTION
**Pull request by ds-jkreutzfeld**

### CODE REVIEW

File: irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        // then
+        given().ignoreException(EntityNotFoundException.class)
+           .await()
+           .atMost(10, TimeUnit.SECONDS)
+           .until(() -> getRelationshipsSize(registeredJob.getJobId()), equalTo(expectedRelationshipsSizeFullTree));
+    }
+
+    @Test
+    @Disabled // its not consistent before TRI-390 not resolved
```
Comment (ds-jhartmann) [inline irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java:62]: Sonar marks this as a Code Smell unless the comment is provided in the format @Disabled("Comment")

**Review by ds-jhartmann — APPROVED** [review]

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java [hunk 2]
```diff
@@ -110,31 +111,77 @@ public Job cancelJobById(final @NonNull UUID jobId) {
...
+
+            } else if (includePartialResults) {
+                final var container = retrievePartialResults(multiJob);
+                relationships.addAll(convert(container.getAssemblyPartRelationships()));
+                tombstones.addAll(container.getTombstones());
+
             }
-            return Jobs.builder().job(multiJob.getJob()).relationships(relationships).tombstones(tombstones).build();
+
+            return Jobs.builder().job(multiJob.getJob()).relationships(relationships).build();
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java:134]: Missing tombstones()

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java [hunk 3]
```diff
@@ -110,31 +111,77 @@ public Job cancelJobById(final @NonNull UUID jobId) {
     }
 
     @Override
-    public Jobs getJobForJobId(final UUID jobId) {
+    public Jobs getJobForJobId(final UUID jobId, final boolean includePartialResults) {
         final Optional<MultiTransferJob> multiTransferJob = jobStore.find(jobId.toString());
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java:115]: Maybe we can add a log somewhere in this method for tracking/metrics purpose?

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java [hunk 4]
```diff
@@ -110,31 +111,77 @@ public Job cancelJobById(final @NonNull UUID jobId) {
...
+
+            } catch (BlobPersistenceException e) {
+                log.error("Unable to read transfer result", e);
+            }
+        }
+        return ItemContainer.builder().assemblyPartRelationships(relationships).tombstones(tombstones).build();
+    }
+
+    private ItemContainer toItemContainer(final byte[] blob) {
+        return new JsonUtil().fromString(new String(blob, StandardCharsets.UTF_8), ItemContainer.class);
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java:166]: In general I think we should create a task in Jira to change a little bit the way we are using ObjectMapper.
As far as I remember creation of ObjectMapper is 'expensive', and we are doing this in few places with every request.
My proposal is to make JsonUtil a spring bean, and inject this whenever its needed, eg:
```
@Bean
    public JsonUtil jsonUtil(ObjectMapper objectMapper) {
        objectMapper.registerModule(new Jdk8Module());
        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
        return new JsonUtil(objectMapper);
    }
```
In above we are reusing objectMapper provided by spring already. What do u think?

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java (hunk 2 above)
Comment (ds-jkreutzfeld) [reply irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java]: Oops :)

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java (hunk 3 above)
Comment (ds-jkreutzfeld) [reply irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java:117]: Added logging 👍

**Review by ds-ext-kmassalski — APPROVED** [review]

**Pull request merged**